cms/static/js/views/import.js

```diff
@@ -2,241 +2,306 @@
-            if (CourseImport.stopGetStatus) { return ;}
-
-            if (currentStage === 4) {
-                // Succeeded
-                CourseImport.displayFinishedImport();
-                $('.view-import .choose-file-button').html(gettext("Choose new file")).show();
-            } else if (currentStage < 0) {
-                // Failed
-                var errMsg = gettext("Error importing course");
-                var failedStage = Math.abs(currentStage);
-                CourseImport.stageError(failedStage, errMsg);
-                $('.view-import .choose-file-button').html(gettext("Choose new file")).show();
-            } else {
-                // In progress
-                updateStage(currentStage);
+        var updateFeedbackList = function (currStageMsg) {
+            var $checkmark = $dom.successStage.find('.icon');
+            currStageMsg = currStageMsg || '';
+
+            function completeStage(stage) {
+                $(stage)
+                    .removeClass("is-not-started is-started")
+                    .addClass("is-complete");
             }
 
-            var time = timeout || 1000;
-            $.getJSON(url,
-                function (data) {
-                    setTimeout(function () {
-                        getStatus(url, time, data.ImportStatus);
-                    }, time);
-                }
-            );
+            function resetStage(stage) {
+                $(stage)
+                    .removeClass("is-complete is-started has-error")
+                    .addClass("is-not-started")
+                    .find('p.error').remove().end()
+                    .find('p.copy').show();
+            }
+
+            switch (current.state) {
+                case STATE.READY:
+                    _.map($dom.stages, resetStage);
+
+                    break;
+
+                case STATE.IN_PROGRESS:
+                    var $prev = $dom.stages.slice(0, current.stage);
+                    var $curr = $dom.stages.eq(current.stage);
+
+                    _.map($prev, completeStage);
+                    $curr.removeClass("is-not-started").addClass("is-started");
+
+                    break;
+
+                case STATE.SUCCESS:
+                    var successUnix = CourseImport.storedImport().date;
+                    var date = moment(successUnix).utc().format('MM/DD/YYYY');
+                    var time = moment(successUnix).utc().format('HH:mm');
+
+                    _.map($dom.stages, completeStage);
+
+                    $dom.successStage
+                        .find('.item-progresspoint-success-date')
+                        .html('(' + date + ' at ' + time + ' UTC)');
+
+                    break;
+
+                case STATE.ERROR:
+                    // Make all stages up to, and including, the error stage 'complete'.
+                    var $prev = $dom.stages.slice(0, current.stage + 1);
+                    var $curr = $dom.stages.eq(current.stage);
```

Comment:
@AkA84 nit: `$prev` and `$curr` are reported as multiple declarations (at least by PyCharm's built-in js inspector). They're not of course, but it might make sense to declare them outside of `switch` block just to make js linters happy.